### PR TITLE
run full test matrix only on PRs to the master branch,

### PR DESCRIPTION
**Pull request by mashehu**
update supported python version target to 3.12

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -28,12 +28,11 @@ jobs:
     runs-on: ${{ matrix.runner }}
     strategy:
       matrix:
-        python-version: ["3.8", "3.11"]
+        python-version: ["3.8", "3.12"]
```
Comment (mirpedrol) [inline .github/workflows/pytest.yml:31]: Should we also update the version in all actions from 3.11 to 3.12?

**Review by mirpedrol — APPROVED** [review]

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (mashehu) [reply .github/workflows/pytest.yml:31]: ah, true

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -43,6 +42,9 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           cache: "pip"
+        # run full test matrix only on PRs to the master branch
+        if: github.ref == 'refs/heads/master' || matrix.runner == 'ubuntu-20.04' && matrix.python-version == '3.8'
```
Comment (fabianegli) [inline .github/workflows/pytest.yml:46]: Isn't the idea to skip the job completely instead of only one of the steps?

Comment (mirpedrol) [reply .github/workflows/pytest.yml:46]: We still want to run the tests once for PRs to dev, with the latest version of ubuntu and python. And test will all the combinations for PRs to master.

**Pull request merged**